external-crates/move/crates/move-package/src/lock_file/schema.rs

```diff
@@ -60,6 +64,16 @@ pub struct Dependency {
     pub digest: Option<String>,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct CompilerToolchain {
+    /// The Move compiler version used to compile this package.
+    #[serde(rename = "compiler-version")]
+    pub compiler_version: String,
+    /// The Move compiler flags used to compile this package.
+    #[serde(rename = "compiler-flags")]
+    pub compiler_flags: BuildConfig,
```

Comment:
Made the decision to store `BuildConfig` fully, I believe that captures the full fidelity of compiler options.